Source/player.cpp

```diff
@@ -4083,72 +4083,63 @@ BOOL __fastcall PosOkPlayer(int pnum, int px, int py)
 	return result;
 }
 
-void __fastcall MakePlrPath(int pnum, int xx, int yy, unsigned char endspace)
+void __fastcall MakePlrPath(int pnum, int xx, int yy, BOOL endspace)
 {
-	int v4; // esi
-	int v5; // ebx
-	int v6; // esi
-	int v7; // edi
-	int v8; // eax
-	int v9; // eax
-	int v10; // ecx
-	int a2; // [esp+Ch] [ebp-4h]
-
-	v4 = pnum;
-	v5 = xx;
-	a2 = pnum;
-	if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( (DWORD)pnum >= MAX_PLRS ) {
 		TermMsg("MakePlrPath: illegal player %d", pnum);
-	v6 = v4;
-	v7 = yy;
-	v8 = plr[v6]._px;
-	plr[v6]._ptargx = v5;
-	plr[v6]._ptargy = yy;
-	if ( v8 != v5 || plr[v6]._py != yy )
-	{
-		v9 = FindPath(PosOkPlayer, a2, v8, plr[v6]._py, v5, yy, plr[v6].walkpath);
-		if ( v9 )
+	}
+
+	plr[pnum]._ptargx = xx;
+	plr[pnum]._ptargy = yy;
+	if ( plr[pnum]._px == xx && plr[pnum]._py == yy ) {
+		return;
+	}
+
+	int path = FindPath(PosOkPlayer, pnum, plr[pnum]._px, plr[pnum]._py, xx, yy, plr[pnum].walkpath);
+	if ( !path ) {
+		return;
+	}
+
+	if ( !endspace ) {
+		path--;
+		switch ( plr[pnum].walkpath[path] )
 		{
-			if ( !endspace )
-			{
-				v10 = plr[v6].walkpath[--v9]; /* *((char *)&plr[v6]._pmode + v9-- + 3) */
-				switch ( v10 )
-				{
-					case 1: // N
-						goto LABEL_12;
-					case 2: // W
-						++v5;
-						break;
-					case 3: // E
-						--v5;
-						break;
-					case 4: // S
-						goto LABEL_15;
-					case 5: // NW
-						++v5;
-						goto LABEL_12;
-					case 6: // NE
-						--v5;
-LABEL_12:
-						v7 = yy + 1;
-						break;
-					case 7: // SE
-						--v5;
-						goto LABEL_15;
-					case 8: // SW
-						++v5;
-LABEL_15:
-						v7 = yy - 1;
-						break;
-					default: // 0/Neutral
-						break;
-				}
-				plr[v6]._ptargx = v5;
-				plr[v6]._ptargy = v7;
-			}
-			plr[v6].walkpath[v9] = -1;
+			case 1: // N

```

Comment:
These cases look like the _walk_path enums.

Reply:
I think your right, I have corrected them as such, this should also address what @mewmew mentioned.